File: src/com/example/javas/easy/E026_BaseballGame.java
/*
EASY PROBLEM (Simplified for 15 min):
Baseball Game - Calculate Score

REQUIREMENTS:
- You're keeping score for a baseball game with special rules
- Given a list of strings ops, apply the following operations:
  - Integer (like "5" or "-2"): Record a new score of that value
  - "+": Record sum of previous two scores
  - "D": Record double of previous score
  - "C": Remove previous score (invalidate)
- Return the sum of all scores after all operations

APPROACH:
- Use a Stack to track valid scores
- Process each operation and update stack accordingly
- Sum all remaining scores in stack

DATA STRUCTURES:
- Stack<Integer>: to track scores

TIME COMPLEXITY: O(n) where n is number of operations
SPACE COMPLEXITY: O(n)
*/

package com.example.javas.easy;

import java.util.*;

public class E026_BaseballGame {
    
    public static int calPoints(String[] operations) {
        // TODO: Implement baseball game scorer
        // 1. Create Stack<Integer>
        // 2. For each operation:
        //    - If "+": push (top + second top) to stack
        //    - If "D": push (top * 2) to stack
        //    - If "C": pop from stack
        //    - Otherwise: push Integer.parseInt(operation) to stack
        // 3. Sum all values in stack and return
        
        return 0;
    }
    
    // TEST CASES
    public static void main(String[] args) {
        String[] ops1 = {"5", "2", "C", "D", "+"};
        System.out.println("Operations [5,2,C,D,+]: " + calPoints(ops1)); // 30
        
        String[] ops2 = {"5", "-2", "4", "C", "D", "9", "+", "+"};
        System.out.println("Operations [5,-2,4,C,D,9,+,+]: " + calPoints(ops2)); // 27
        
        String[] ops3 = {"1", "C"};
        System.out.println("Operations [1,C]: " + calPoints(ops3)); // 0
        
        String[] ops4 = {"1"};
        System.out.println("Operations [1]: " + calPoints(ops4)); // 1
    }
}

File: pythonX/easy/e026_baseball_game.py
"""
E026: Baseball Game - Calculate Score

Summary:
Baseball Game

Requirements:
- You're keeping score for a baseball game with special rules
- Given a list of strings ops, apply the following operations:
  - Integer (like "5" or "-2"): Record a new score of that value
  - "+": Record sum of previous two scores
  - "D": Record double of previous score
  - "C": Remove previous score (invalidate)
- Return the sum of all scores after all operations

Approach:
- Use a list as stack to track valid scores
- Process each operation and update stack accordingly
- Sum all remaining scores in stack

Data Structures:
- list: to track scores (used as stack)

Big O Analysis:
Time Complexity: O(n) where n is number of operations
Space Complexity: O(n)
"""

def cal_points(operations: list[str]) -> int:
    """
    TODO: Implement baseball game scorer
    1. Create list to use as stack
    2. For each operation:
       - If "+": append (stack[-1] + stack[-2]) to stack
       - If "D": append (stack[-1] * 2) to stack
       - If "C": pop from stack
       - Otherwise: append int(operation) to stack
    3. Sum all values in stack and return
    """
    pass


if __name__ == "__main__":
    # Test Cases
    ops1 = ["5", "2", "C", "D", "+"]
    print(f"Operations ['5','2','C','D','+']: {cal_points(ops1)}")  # 30
    
    ops2 = ["5", "-2", "4", "C", "D", "9", "+", "+"]
    print(f"Operations ['5','-2','4','C','D','9','+','+']: {cal_points(ops2)}")  # 27
    
    ops3 = ["1", "C"]
    print(f"Operations ['1','C']: {cal_points(ops3)}")  # 0
    
    ops4 = ["1"]
    print(f"Operations ['1']: {cal_points(ops4)}")  # 1
